### DISCUSSION TITLE
How to clear autocomplete and up key history?

### DISCUSSION THREAD

**Discussion by manujchandra** [xonsh/xonsh#4928] [Q&A] [answered]
Hi,

I am using Xonsh version 0.10.1 with Alacrity terminal.

I recently processed some OTP codes URI on the terminal using python qr library. It contains sensitive information regarding token IDS.

After that, I issued the commands history flush and history clear. But there are two problems:

1) When I press the up arrow, it refills the previous command which contains the sensitive information and it keeps going back in history.
2) When I write qr it tries to autocompletes the full otp uri with sensitive codes (in gray color).

How do I delete the history in such a manner that there is no autocomplete and also no history when I press the up button. Because that data should not get flashed during screen sharing accidentally.

Thanks.

![Screenshot_2022-08-28_19-49-14](https://user-images.githubusercontent.com/15191493/187078662-449c6b3e-a542-4ac1-bf02-20c14531f2a2.png)

**Comment by anki-code** [comment]
If you are using [SQLite history backend](https://github.com/anki-code/xontrib-rc-awesome/blob/86d29c622c92adac201a1958aa27b1750588ad88/xontrib/rc_awesome.xsh#L26-L28) you can delete the commands from history as ease as possible by using SQL ([example](https://github.com/anki-code/xonsh-cheatsheet#history)). After deleting and restarting the shell the command will disappeared.

Also just for information the prompt-toolkit feature about auto suggest described here - https://python-prompt-toolkit.readthedocs.io/en/master/pages/asking_for_input.html#auto-suggestion

**Comment by manujchandra** [comment, marked as answer]
When I write history info, it says backend is json not sqlite

```
backend: json
sessionid: XXX
filename: /home/manuj/.local/share/xonsh/history_json/xonsh-fb36ba52-669d-4f54-b012-14557c0b6fb2.json
length: 0
buffersize: 100
bufferlength: 0
gc options: (8128, 'commands')
gc_last_size: (8128, 'commands')
```

**Reply by anki-code** [reply]
Yes. By default xonsh uses json. You should add to `~/.xonshrc`:
```python
$XONSH_HISTORY_BACKEND = 'sqlite'
```
and you can do:
```xsh
sqlite3 $XONSH_HISTORY_FILE "DELETE FROM xonsh_history WHERE inp LIKE 'curl%'"
```

**Reply by gforsyth** [reply]
If you navigate to the `~/.local/share/xonsh/history_json` directory you can use `grep` or `ag` or `rg` or whatever to find which of the `json` files has the sensitive info in it and then nuke that `json` file.  It's a bit coarse, but that will ensure that it doesn't show up in your autocomplete and up-key history

**Reply by manujchandra** [reply]
> If you navigate to the `~/.local/share/xonsh/history_json` directory you can use `grep` or `ag` or `rg` or whatever to find which of the `json` files has the sensitive info in it and then nuke that `json` file. It's a bit coarse, but that will ensure that it doesn't show up in your autocomplete and up-key history

This solved the issue. Thanks a lot.
